src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -2480,3 +2474,56 @@ std::string dbDotDecode(std::string s)
   std::replace(s.begin(), s.end(), ESCAPE_1_ENCODED, ESCAPE_1_DECODED);
   return s;
 }
+
+
+
+/* ****************************************************************************
+*
+* subscriptionsTreat -
+*
+* Lookup all subscriptions in the database and call a treat function for each
+*/
+void subscriptionsTreat(std::string tenant, OtisTreatFunction treatFunction)
+{
+  BSONObj                   query;
+  DBClientBase*             connection = getMongoConnection();
+  auto_ptr<DBClientCursor>  cursor;
+
+  try
+  {
+    cursor = connection->query(getSubscribeContextCollectionName(tenant).c_str(), query);
```

Comment:
True, the query is probably faster without extracting 'garbage' from the DB.
I might need some help in constructing the query for mongo ...